docker_mcp/services/container.py

```
@@ -639,45 +663,225 @@
+            self.logger.error(
+                "Failed to check port availability", host_id=host_id, port=port, error=str(e)
+            )
             return {
                 "success": False,
                 "error": f"Port check failed: {str(e)}",
-                "host_id": host_id,
+                HOST_ID: host_id,
                 "port": port,
             }
+
+    async def handle_action(self, action, **params) -> dict[str, Any]:
```

Ruff (0.12.2)

`handle_action` is too complex (26 > 10)

(C901)